beesdoo_product/models/beesdoo_product.py

```diff
@@ -64,12 +64,20 @@ class BeesdooProduct(models.Model):
         translate=True,
     )
 
-    main_seller_id = fields.Many2one(
-        "res.partner",
-        string="Main Seller",
+    top_supplierinfo_id = fields.Many2one(
+        comodel_name="product.supplierinfo",
         compute="_compute_main_seller_id",
         store=True,
     )
+    main_seller_id = fields.Many2one(
+        string="Main Seller",
+        comodel_name="res.partner",
+        related="top_supplierinfo_id.name",
+    )
```

Comment:
The field is named `main_seller_id` but it points to a name, shouldn't it be called `main_seller_id_name` instead (would also be consistent with main_seller_id_product_code) ?

Reply:
You are right. But that's an odoo choice to put a res.partner into a field `name` of a supplierinfo object. Strange ideed, but cannot change that.

Reply:
supplierinfo object is a glue object between a partner (which is the seller) and a product.

Reply:
Using `main_seller_id_name` will have no sens because we will expect this variable to contain a string that is the name of the supplier which is not. Because in `name` there is a res.partner. :)

Reply:
I'm not sure to be clear. ^^

Reply:
OK, no it's clear :) Thanks for the explanation